tmt/steps/discover/fmf.py

```diff
@@ -562,7 +568,8 @@ def do_the_discovery(self, path: Optional[Path] = None) -> None:
         tree = tmt.Tree(
             logger=self._logger,
             path=tree_path,
-            fmf_context=self.step.plan._fmf_context)
+            fmf_context=self.step.plan._fmf_context,
+            additional_rules=self.get('adjust-tests'))
```

Comment:
`self.data.adjust` should be preferred over `self.get()` as `self.get()` can't handle the type annotations.

Reply:
Yes, that should work, type checkers should grab the right type (because `self.data` is a dataclass), and match it against the `additional_rules` signature.